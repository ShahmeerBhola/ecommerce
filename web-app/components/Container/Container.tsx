import React from 'react';

export type Props = {
  className?: string;
  paddingY?: string;
  paddingX?: string;
  bgColor?: Styles.BackgroundColors;
  innerContainerClassName?: string;
};

const Container: React.FunctionComponent<Props> = ({
  children,
  className,
  paddingY = 'md:py-10 py-5',
  paddingX = 'px-5',
  bgColor,
  innerContainerClassName = '',
}) => {
  const baseClasses = className ? `${className} ` : '';
  const innerContainerClasses = `container mx-auto ${paddingX} ${paddingY} ${innerContainerClassName}`.trim();

  if (bgColor) {
    return (
      <div className={`${baseClasses}${bgColor}`}>
        <div className={innerContainerClasses}>{children}</div>
      </div>
    );
  }
  return <div className={`${baseClasses}${innerContainerClasses}`}>{children}</div>;
};

export default Container;
