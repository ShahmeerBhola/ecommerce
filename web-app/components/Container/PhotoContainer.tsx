import React from 'react';
import Container from './Container';
import PhotoBackground from '../PhotoBackground';

type Props = {
    image?: string;
    className?: string;
    backgroundCover?: boolean;
    backgroundClassName?: string;
    backgroundStyles?: React.CSSProperties;
    containerPaddingY?: string;
};

const PhotoContainer: React.FunctionComponent<Props> = ({
    children,
    image,
    className = '',
    backgroundCover,
    backgroundClassName,
    backgroundStyles,
    containerPaddingY,
}) => {
    return (
        <PhotoBackground
            image={image ? `/images/${image}` : undefined}
            cover={backgroundCover}
            className={backgroundClassName}
            styles={backgroundStyles}
        >
            <Container className={className} paddingY={containerPaddingY}>
                {children}
            </Container>
        </PhotoBackground>
    );
};

export default PhotoContainer;
