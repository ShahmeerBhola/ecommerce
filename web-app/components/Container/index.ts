import Container from './Container';
import PhotoContainer from './PhotoContainer';

export default Container;
export { PhotoContainer };
