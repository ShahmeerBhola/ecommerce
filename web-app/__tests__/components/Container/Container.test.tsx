import React from 'react';
import renderer from 'react-test-renderer';
import Container from '../../../components/Container/Container';

describe('Container component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <Container>
          <p>Hello</p>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly when bgColor prop is set', () => {
    const tree = renderer
      .create(
        <Container bgColor="gray-75">
          <p>Hello</p>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default props', () => {
    const tree = renderer
      .create(
        <Container className="hello-there" minHeight="500" paddingY="25">
          <p>Hello</p>
        </Container>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
