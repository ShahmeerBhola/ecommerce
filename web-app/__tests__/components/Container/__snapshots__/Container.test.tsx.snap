// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Container component it renders properly 1`] = `
<div
  className="container mx-auto px-5 md:py-20 py-5 min-h-80"
>
  <p>
    Hello
  </p>
</div>
`;

exports[`Container component it renders properly when bgColor prop is set 1`] = `
<div
  className="bg-gray-75"
>
  <div
    className="container mx-auto px-5 md:py-20 py-5 min-h-80"
  >
    <p>
      Hello
    </p>
  </div>
</div>
`;

exports[`Container component it renders properly with non-default props 1`] = `
<div
  className="container mx-auto px-5 25 500 hello-there"
>
  <p>
    Hello
  </p>
</div>
`;
